presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/util/ThriftExceptions.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.util;
+
+import com.facebook.presto.connector.thrift.api.PrestoThriftServiceException;
+import com.facebook.presto.spi.PrestoException;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.drift.TApplicationException;
+import io.airlift.drift.TException;
+import io.airlift.drift.protocol.TTransportException;
+
+import static com.facebook.presto.connector.thrift.ThriftErrorCode.THRIFT_SERVICE_CONNECTION_ERROR;
+import static com.facebook.presto.connector.thrift.ThriftErrorCode.THRIFT_SERVICE_GENERIC_REMOTE_ERROR;
+import static com.facebook.presto.connector.thrift.ThriftErrorCode.THRIFT_SERVICE_NO_AVAILABLE_HOSTS;
+import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static com.google.common.util.concurrent.Futures.catchingAsync;
+import static com.google.common.util.concurrent.Futures.immediateFailedFuture;
+
+public final class ThriftExceptions
+{
+    private ThriftExceptions() {}
+
+    public static PrestoException thriftException(Exception e)
```

Comment:
maybe name this something else like `toPrestoException` or `thriftToPrestoException`?